### PR TITLE
Uwp6.2: Port size on disk related fixes

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
```
git cherry-pick 6867b6fc9327d2f9c3e36dd5889a0eb9c3991da5
git cherry-pick e5720ae475b90710586b734f149df2ded318cf73
git cherry-pick 5898cdd5b835fde14bb3c25c39755725de02f2b9
git cherry-pick d21453e1cfa2e2546deccbd3a4be46341e4ba5d1
git cherry-pick 39e96cd8e3b97f8b1a5fce86211cf8fec7ea478a
```

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
@dotnet-bot test Windows x64 Debug Build

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlDataContract.cs [hunk 1]
```diff
@@ -379,6 +380,7 @@ private bool RequiresMemberAccessForCreate(SecurityException securityException)
         }
 #endif
 
+        [RemovableFeature(ReflectionBasedSerializationFeature.Name)]
```
Comment (zamont) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlDataContract.cs]: Is it going to be an issue that this RemovableFeatureAttribute is not ifdef'ed for only uapaot?

File: src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs [hunk 2]
```diff
@@ -33,6 +34,27 @@ internal partial class XmlDownloadManager
...
+                return result;
+            }
+        }
+
+        // This method is marked Removable because WebRequest has a lot of dependencies that will bloat
+        // self-contained distributions of .NET Apps.
+        // This code is statically reachable from any place that uses XmlReaderSettings (i.e. every app that
+        // does something XML related is going to have this in their transitive call graph). People rarely need
+        // this functionality though.
+        [RemovableFeature("System.Xml.XmlUrlResolver.NonFileUrlSupport")]
```
Comment (zamont) [inline src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs]: Should this RemovableFeatureAttribute also be in an ifdef?

**Comment by zamont** [comment]
The Linux x64 Release Build doesn't show any test failures in Jenkins, so I'm not sure why it returned as a failure.

**Comment by zamont** [comment]
LGTM besides my two comments

**Comment by zamont** [comment]
Turns out the Fedora.26.Amd configuration has reached end-of-life. I'll put that change on top of Morgan's existing PR and I hope that will be merged on Monday

**Comment by zamont** [comment]
Morgan's PR was clean and is here for reference: #30886

**Pull request merged**